Initial docs for metrics

markpollack opened the pull request:
Note that UI and arch diagram are not included.  That can be in another PR.  

Fixes #1288

markpollack commented:
https://docs.google.com/a/pivotal.io/presentation/d/1D1blpqouCVpk8tBCYX9JPB69MEXK-g0HwqiWiOgOjxE/edit?usp=sharing can be used for arch diagram.

cppwfs commented on spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc:
```diff
@@ -1,4 +1,4 @@
-[[configuration]]
+o[[configuration]]
```
o added incorrectly

markpollack replied:
strange, I don't see this 'o' in my branch....

markpollack commented:
added high level arch diagram

ghillert commented:
@markpollack I wonder if it would be a good idea to add a quick example (with cut & paste steps) on how to get this working, e.g.:

```
$ java -jar target/metrics-collector-rabbit-1.0.0.BUILD-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=metrics --spring.cloud.stream.default.contentType=application/json --security.basic.enabled=false --management.security.enabled=false

$ java -jar spring-cloud-dataflow-server-local/target/spring-cloud-dataflow-server-local-1.2.0.BUILD-SNAPSHOT.jar --spring.cloud.dataflow.metrics.collector.url=http://localhost:8080/collector/metrics/streams

dataflow:>app register --name time120RS --type source --uri maven://org.springframework.cloud.stream.app:time-source-rabbit:1.2.0.RELEASE

dataflow:>app register --name log120RS --type sink --uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.2.0.RELEASE

stream create --name foostream --definition "time120RS |log120RS"

stream deploy --name foostream --properties "deployer.*.count=2,app.*.security.basic.enabled=false,app.*.management.security.enabled=false,app.*.spring.cloud.stream.bindings.applicationMetrics.destination=metrics"
```

And combine that with a screenshot of the UI so that (new) users get an idea of what they ultimately will see.

markpollack commented:
Added added a link to the Metrics Collector project page that should have the url coordinates to do a `wget` of each server and how to run the server.

Added the app register command so that users get the right Spring Cloud Stream app versions that have metrics.

markpollack commented:
Squashed and merged as c7ed91fee5517eef3d475061285d6dea483e5397
